apis/v1alpha1/secret.go

```diff
@@ -0,0 +1,199 @@
+	//     use PutSecretValue to create a new version.
+	//
+	// This value becomes the VersionId of the new version.
+	ClientRequestToken *string `json:"clientRequestToken,omitempty"`
+	// The description of the secret.
+	Description *string `json:"description,omitempty"`
+	// Specifies whether to overwrite a secret with the same name in the destination
+	// Region. By default, secrets aren't overwritten.
+	ForceOverwriteReplicaSecret *bool `json:"forceOverwriteReplicaSecret,omitempty"`
+	// The ARN, key ID, or alias of the KMS key that Secrets Manager uses to encrypt
+	// the secret value in the secret. An alias is always prefixed by alias/, for
+	// example alias/aws/secretsmanager. For more information, see About aliases
+	// (https://docs.aws.amazon.com/kms/latest/developerguide/alias-about.html).
+	//
+	// To use a KMS key in a different account, use the key ARN or the alias ARN.
+	//
+	// If you don't specify this value, then Secrets Manager uses the key aws/secretsmanager.
+	// If that key doesn't yet exist, then Secrets Manager creates it for you automatically
+	// the first time it encrypts the secret value.
+	//
+	// If the secret is in a different Amazon Web Services account from the credentials
+	// calling the API, then you can't use aws/secretsmanager to encrypt the secret,
+	// and you must create and use a customer managed KMS key.
+	KMSKeyID *string `json:"kmsKeyID,omitempty"`
+	// The name of the new secret.
+	//
+	// The secret name can contain ASCII letters, numbers, and the following characters:
+	// /_+=.@-
+	//
+	// Do not end your secret name with a hyphen followed by six characters. If
+	// you do so, you risk confusion and unexpected results when searching for a
+	// secret by partial ARN. Secrets Manager automatically adds a hyphen and six
+	// random characters after the secret name at the end of the ARN.
+	// +kubebuilder:validation:Required
+	Name *string `json:"name"`
+	// The binary data to encrypt and store in the new version of the secret. We
+	// recommend that you store your binary data in a file and then pass the contents
+	// of the file as a parameter.
+	//
+	// Either SecretString or SecretBinary must have a value, but not both.
+	//
+	// This parameter is not available in the Secrets Manager console.
+	SecretBinary []byte `json:"secretBinary,omitempty"`
+	// The text data to encrypt and store in this new version of the secret. We
+	// recommend you use a JSON structure of key/value pairs for your secret value.
+	//
+	// Either SecretString or SecretBinary must have a value, but not both.
+	//
+	// If you create a secret by using the Secrets Manager console then Secrets
+	// Manager puts the protected secret text in only the SecretString parameter.
+	// The Secrets Manager console stores the information as a JSON structure of
+	// key/value pairs that a Lambda rotation function can parse.
+	SecretString *string `json:"secretString,omitempty"`
```

Comment:
I'm a bit concerned about these being written it plaintext in the CR Spec. It might be a better idea for these to be references to K8s Secrets instead?

Reply:
good idea!  let me investigate how to do it

Reply:
This should be a 1 line change. See - https://github.com/aws-controllers-k8s/rds-controller/blob/main/generator.yaml#L111

I still feel strongly that writing secrets in plaintext CRs is not the right move, even for v0.0.1

Reply:
- add SecretString as k8s secret
- remove SecretBinary